### ISSUE TITLE
Flatpak: Request to run in background before minimizing to tray

### ISSUE THREAD

**Issue by mak448a** [keepassxreboot/keepassxc#9509]
## Overview
Keepassxc crashes a few seconds after being minimized to tray.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Enable Minimize instead of app exit, show a system tray icon, & hide window to system tray when minimized.
2. Close the app with `Alt`+`F4`
3. CRASH!

## Expected Behavior
It stays in the tray

## Actual Behavior
It crashes after in the tray for a few seconds

KeePassXC - Version 2.7.5
Revision: 9d0537b
Distribution: Flatpak

Qt 5.15.9
Debugging mode is disabled.

Operating system: KDE Flatpak runtime
CPU architecture: x86_64
Kernel: linux 6.3.4-201.fc38.x86_64

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.19.3

Operating System: Linux
Desktop Env: KDE
Windowing System: Wayland

**Comment by droidmonkey** [comment]
I cannot replicate this at all. Tried in the same environment: Fedora 37 // KDE // Flatpak install. Please uninstall and reinstall keepassxc.

**Comment by mak448a** [comment]
Got it, I'll uninstall and reinstall.

**Comment by em2943** [comment]
have you got it working?

**Comment by mak448a** [comment]
Yes. Just make sure to remove `~/.var/app/org.keepassxc.KeePassXC` when uninstalling.

**Comment by em2943** [comment]
when I install the flatpak the sys tray icon was invisible. at least you got it working.

**Comment by mak448a** [comment]
@folfgirl You need to enable it in the settings

**Comment by mak448a** [comment]
![Screenshot_20230605_180157](https://github.com/keepassxreboot/keepassxc/assets/94062293/eefc881b-6c43-4fb7-b9e2-8eb0fc7b251a)

**Comment by em2943** [comment]
I have but the systray icon is invisible

**Comment by mak448a** [comment]
Try setting the tray icon type (in the screenshot) to something else

**Comment by dhanak** [comment]
KeepassXC worked for me just fine until recently, but today, it started demonstrating this very same behavior. I haven't changed any configuration, I haven't upgraded the system or anything. It runs just fine in the foreground, but when I minimize it to the tray, after a few seconds, it silently exits. It doesn't print anything on the console, either. After deleting the `~/.var/app/org.keepassxc.KeePassXC` directory, launching a brand-new instance, and then setting the "minimize to system tray" to on and minimizing the app ends with the same crash. Changing the tray icon type also doesn't solve anything. I also tried upgrading, to no avail.

This is pretty annoying.

KeePassXC - Version 2.7.6
Revision: dd21def
Distribution: Flatpak

Qt 5.15.10
Debugging mode is disabled.

Operating system: KDE Flatpak runtime
CPU architecture: x86_64
Kernel: linux 6.1.0-13-amd64

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 3.2.0

**Comment by droidmonkey** [comment]
Can you try to capture a stack trace? 

https://docs.flatpak.org/en/latest/debugging.html

**Comment by dhanak** [comment]
I installed the devel SDK as instructed, plus the .Debug package for keepassxc, then started flatpak with `flatpak --command-=sh --devel org.keepassxc.KeePassXC`, and got the command prompt as expected. Then something strange happened: this image itself exited after 10 seconds or so, without me doing anything! keepassxc wasn't even running.

When I try the same with another flatpak, e.g., `com.skype.Client`, the same thing happens.

**Comment by dhanak** [comment]
Argh. I got this figured out. It was Gnome all along. Unknown to me, there is a feature in Gnome which makes apps running in the background require a special permission. It is probably on by default, but there is a notification popup each time the app goes in the background, and somehow, I probably managed to switch this to off accidentally without noticing. And then, Gnome just kills my keepassxc as soon as it goes in background.

![Screenshot from 2023-11-20 15-40-58](https://github.com/keepassxreboot/keepassxc/assets/55064/4e4c5298-ca3c-4ba3-8948-3545cfc8f929)

Changing that toggle back to enabled solved the issue.

**Comment by droidmonkey** [comment]
That is a crazy "feature"

**Comment by mak448a** [comment]
@dhanak Thank you! It was toggled off for me too, I had to enable it through flatseal

**Comment by dhanak** [comment]
See flatpak/xdg-desktop-portal#901 and flatpak/flatpak#5427 for more explanation.

**Comment by droidmonkey** [comment]
What a calamity. Since GNOME no longer formally recognizes a system tray, it must declare everything without a window a background app even though we also declare a tray presence. This likely doesn't happen on KDE because it recognizes a tray. 

They also said this is no longer the default behavior after desktop portal 1.16 so I hope this just goes away.

**Comment by droidmonkey** [comment]
What a calamity. I guess we need to add the background permission to our flatpak now.

**Comment by droidmonkey** [comment]
Looks like we need to explicitly request permission prior to the first minimize to tray: https://docs.flatpak.org/en/latest/portal-api-reference.html#gdbus-org.freedesktop.portal.Background

**Comment by droidmonkey** [comment]
> xdg-desktop-portal >= v1.17 allows apps to run in the background by default

I'm considering this OBE
